src/Patterns/Scd2Pattern.php

```diff
@@ -81,51 +64,44 @@ protected function getTemplateVariables(): array
         ];
     }
 
-    private function getInputColumns(): array
-    {
-        return array_merge($this->getParameters()->getPrimaryKey(), $this->getParameters()->getMonitoredParameters());
-    }
-
     private function getSnapshotPrimaryKeyParts(): array
     {
         // All snapshot columns are lower
-        return $this->columnsToLower(
+        $columns = $this->getColumnsWithDefinition(
             array_merge($this->getParameters()->getPrimaryKey(), [$this->getParameters()->getStartDateName()])
         );
-    }
 
-    private function getSnapshotInputColumns(): array
-    {
-        return $this->getParameters()->getUppercaseColumns()
-            ? $this->columnsToUpper($this->getInputColumns())
-            : $this->columnsToLower($this->getInputColumns());
+        return $this->columnsToLower($columns);
     }
 
-    private function getSnapshotSpecialColumns(): array
+    protected function getSnapshotSpecialColumns(): array
     {
         return array_values($this->getSnapshotSpecialColumnsWithKeys());
     }
 
     private function getSnapshotSpecialColumnsWithKeys(): array
     {
-        $columns['startDate'] = $this->getParameters()->getStartDateName();
-        $columns['endDate'] = $this->getParameters()->getEndDateName();
-        $columns['actual'] = $this->getParameters()->getActualName();
+        $columns['startDate'] = [
+            'name' => $this->getParameters()->getStartDateName(),
+            'definition' => ['type' => $this->getParameters()->useDatetime() ? 'DATETIME' : 'DATE'],
+        ];
+
+        $columns['endDate'] = [
+            'name' => $this->getParameters()->getEndDateName(),
+            'definition' => ['type' => $this->getParameters()->useDatetime() ? 'DATETIME' : 'DATE'],
```

Comment:
tady to definition se pak používá jen pro typované tabulky, pro netypované to tu je pak navíc - ale v tuhle chvíli nevím jaká se bude zakládat